### ISSUE TITLE
Template Parser Error: Multiple @case statements in @switch block fail to parse

### ISSUE THREAD

**Issue by stephenlautier** [angular-eslint/angular-eslint#2869]
# Template Parser Error: Multiple @case statements in @switch block fail to parse

## Description and reproduction of the issue

When using Angular's native control flow `@switch` statement with multiple `@case` statements that share the same block body (a valid Angular syntax), ESLint fails to parse the template with the error:

```
Parsing error: Incomplete block "case". If you meant to write the @ character, you should use the "&#64;" HTML entity instead
```

### Reproduction

Create a new Angular 21 project and add ESLint:

```bash
ng new my-app
ng add angular-eslint
```

Add the following code to any component template:

```html
@switch(theme()) {
  @case("dark")
  @case("light") {
    <h1>Light/Dark</h1>
  }
}
```

Run ESLint:

```bash
npm run lint
```

**Expected behavior:** Lint should pass without errors, as this is valid Angular syntax where multiple cases can share the same block.

**Actual behavior:** ESLint fails with parsing error:
```
error  Parsing error: Incomplete block "case". If you meant to write the @ character, you should use the "&#64;" HTML entity instead
```

## Versions

| package                            | version |
| ---------------------------------- | ------- |
| `@angular-eslint/template-parser`  | 21.1.0  |
| `angular-eslint`                   | 21.1.0  |
| `ESLint`                           | 9.39.1  |
| `node`                             | 24.11.1 |

### Output from `npx ng version`:

```
Angular CLI       : 21.1.1
Angular           : 21.1.1
Node.js           : 24.11.1
Package Manager   : pnpm 9.15.4
Operating System  : win32 x64

┌───────────────────────────┬───────────────────┬───────────────────┐
│ Package                   │ Installed Version │ Requested Version │
├───────────────────────────┼───────────────────┼───────────────────┤
│ @angular/build            │ 21.1.1            │ ^21.1.1           │
│ @angular/cli              │ 21.1.1            │ ^21.1.1           │
│ @angular/common           │ 21.1.1            │ ^21.1.0           │
│ @angular/compiler         │ 21.1.1            │ ^21.1.0           │
│ @angular/compiler-cli     │ 21.1.1            │ ^21.1.0           │
│ @angular/core             │ 21.1.1            │ ^21.1.0           │
│ @angular/forms            │ 21.1.1            │ ^21.1.0           │
│ @angular/platform-browser │ 21.1.1            │ ^21.1.0           │
│ @angular/router           │ 21.1.1            │ ^21.1.0           │
│ rxjs                      │ 7.8.2             │ ~7.8.0            │
│ typescript                │ 5.9.3             │ ~5.9.2            │
│ vitest                    │ 4.0.18            │ ^4.0.8            │
└───────────────────────────┴───────────────────┴───────────────────┘
```

### Relevant dependencies from `package.json`:

```json
"devDependencies": {
  "@angular-eslint/builder": "21.1.0",
  "@angular-eslint/eslint-plugin": "21.1.0",
  "@angular-eslint/eslint-plugin-template": "21.1.0",
  "@angular-eslint/schematics": "21.1.0",
  "@angular-eslint/template-parser": "21.1.0",
  "angular-eslint": "21.1.0",
  "eslint": "^9.39.1",
  "typescript-eslint": "8.47.0"
}
```

- [x] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

## Additional Context

This syntax is valid according to Angular documentation and works correctly at runtime. Multiple `@case` statements without individual bodies should share the next block that follows them. This is similar to fall-through behavior in JavaScript switch statements.

Reference: The Angular compiler accepts this syntax without issues, and the application runs correctly.

**Comment by reduckted** [comment]
That feature was added in Angular 21.1.0
- https://github.com/angular/angular/releases/tag/v21.1.0
- https://github.com/angular/angular/commit/640693da8e667c015662246152236585d9b24e7f

It would be supported in angular-eslint via https://github.com/angular-eslint/angular-eslint/pull/2862

**Comment by JamesHenry** [comment]
Should be fixed in https://github.com/angular-eslint/angular-eslint/releases/tag/v21.2.0
